app/src/main/java/me/guillaumin/android/osmtracker/gpx/TrackFileProvider.java

```diff
@@ -0,0 +1,121 @@
+package me.guillaumin.android.osmtracker.gpx;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.AbstractCursor;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
+ */
+
+public final class TrackFileProvider extends ContentProvider {
+    private static final String TAG = ExportToTempFileTask.class.getSimpleName();
+
+    @Override
+    public boolean onCreate() {
+        return false;
+    }
+
+    @Override
+    public Cursor query(Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder) {
+        return new AbstractCursor() {
+            @Override
+            public int getCount() {
+                return 0;
+            }
+
+            @Override
+            public String[] getColumnNames() {
+                return new String[0];
+            }
+
+            @Override
+            public String getString(int column) {
+                return null;
+            }
+
+            @Override
+            public short getShort(int column) {
+                return 0;
+            }
+
+            @Override
+            public int getInt(int column) {
+                return 0;
+            }
+
+            @Override
+            public long getLong(int column) {
+                return 0;
+            }
+
+            @Override
+            public float getFloat(int column) {
+                return 0;
+            }
+
+            @Override
+            public double getDouble(int column) {
+                return 0;
+            }
+
+            @Override
+            public boolean isNull(int column) {
+                return false;
+            }
+        };
+    }
+
+    @Override
+    public String getType(Uri uri) {
```

Comment:
I think here and in `getStreamTypes()` this can also be ZIP? It seem to have an effect, because when I share a ZIP to Google Drive the file in Google Drive ends up being not recognized as an archive, even if it has the ZIP extension.

![drive1](https://user-images.githubusercontent.com/762512/28250416-70988a34-6a1c-11e7-8a88-b7209d1192d9.png)

Also when I open it in Google Drive it shows it as a text or xml file, rather than opening the ZIP archive.

![drive2](https://user-images.githubusercontent.com/762512/28250418-7a7b1878-6a1c-11e7-8b8b-19436395b0ee.png)